ci_tools/list_docs_tovalidate.py

```diff
@@ -0,0 +1,53 @@
+""" Script to list the objects that should be numpydoc validated
+"""
+import ast
+from _ast import FunctionDef, ClassDef, Module
+import argparse
+from git_evaluation_tools import get_diff_as_json as gdj
+
+parser = argparse.ArgumentParser(description='List the objects with docstrings in the files provided')
+parser.add_argument('gitdiff', metavar='gitdiff', type=str,
+                        help='Output of git diff between PR branch and base branch')
+parser.add_argument('output', metavar='output', type=str,
+                        help='File to save the output to')
+args = parser.parse_args()
+
+results = gdj(args.gitdiff)
+changes = {}
+for file, upds in results.items():
+    if file.startswith('pyccel/'):
+        for line_no in upds['addition']:
+            if file in changes:
+                changes[file].append(int(line_no))
+            else:
+                changes[file] = []
+                changes[file].append(int(line_no))
+
+objects = []
+for file, line_nos in changes.items():
+    with open(file,'r', encoding="utf-8") as f:
+        tree = ast.parse(f.read())
+    prefix = file[:-3].replace('/', '.')
+    for node in ast.walk(tree):
+        if isinstance(node, (FunctionDef, ClassDef, Module)):
+            if isinstance(node, Module):
+                objects.append(prefix)
+            if isinstance(node, FunctionDef):
+                if hasattr(node, "parent"):
+                    if any((node.lineno <= x <= node.end_lineno
+                            for x in line_nos)):
+                        objects.append('.'.join([prefix,node.parent.name, node.name]))
+                else:
+                    if any((node.lineno <= x <= node.end_lineno
+                            for x in line_nos)):
+                        objects.append('.'.join([prefix, node.name]))
+            if isinstance(node, ClassDef):
+                if any((node.lineno <= x <= node.end_lineno
+                        for x in line_nos)):
+                    objects.append('.'.join([prefix, node.name]))
+                for child in node.body:
+                    if isinstance(child, ast.FunctionDef):
+                        child.parent = node
```

Comment:
To reduce code duplication and increase readability I suggest:
```suggestion
            if isinstance(node, (FunctionDef, ClassDef)):
                code_changed = any((node.lineno <= x <= node.end_lineno
                            for x in line_nos))
                if code_changed:
                    objects.append('.'.join([prefix, node.name]))
                    
                # Update member function names
                if isinstance(node, ClassDef):
                    for child in node.body:
                        if isinstance(child, ast.FunctionDef):
                            child.name = f'{node.name}.{child.name}'
```
(I have not tested this though so best to double check it works correctly)